include/picolibrary/crc.h

```diff
@@ -472,59 +508,180 @@ class Augmented_Byte_Lookup_Table_Calculator {
+    constexpr Direct_Byte_Lookup_Table_Calculator( Direct_Byte_Lookup_Table_Calculator const & original ) noexcept = default;
+
+    /**
+     * \brief Destructor.
+     */
+    ~Direct_Byte_Lookup_Table_Calculator() noexcept = default;
+
+    /**
+     * \brief Assignment operator.
      *
-     * \return The resulting CRC calculation remainder.
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    constexpr auto operator=( Direct_Byte_Lookup_Table_Calculator && expression ) noexcept
+        -> Direct_Byte_Lookup_Table_Calculator & = default;
+
+    /**
+     * \brief Assignment operator.
+     *
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    constexpr auto operator=( Direct_Byte_Lookup_Table_Calculator const & expression ) noexcept
+        -> Direct_Byte_Lookup_Table_Calculator & = default;
+
+    /**
+     * \copydoc picolibrary::CRC::Calculator_Concept::calculate()
      */
     template<typename Iterator>
-    auto feed( Register remainder, Iterator begin, Iterator end ) const noexcept
+    auto calculate( Iterator begin, Iterator end ) const noexcept -> Register
     {
+        auto remainder = m_initial_remainder;
+
         for ( ; begin != end; ++begin ) {
             auto const processed_input = ( *m_process_input )( *begin );
 
+            auto const i = static_cast<std::uint8_t>(
+                               remainder >> ( std::numeric_limits<Register>::digits
+                                              - std::numeric_limits<std::uint8_t>::digits ) )
+                           ^ processed_input;
+
+            remainder <<= std::numeric_limits<std::uint8_t>::digits;
+
+            remainder ^= m_lookup_table[ i ];
+        } // for
+
+        return ( *m_process_output )( remainder ) ^ m_xor_output;
+    }
+
+  private:
+    /**
+     * \brief Calculation lookup table.
+     */
+    Byte_Lookup_Table<Register> m_lookup_table{};
+
+    /**
+     * \brief Calculation preprocessed initial remainder.
+     */
+    Register m_initial_remainder{};
+
+    /**
+     * \brief Calculation input processor.
+     */
+    Input_Processor m_process_input{};
+
+    /**
+     * \brief Calculation output processor.
+     */
+    Output_Processor<Register> m_process_output{};
+
+    /**
+     * \brief Calculation XOR output value.
+     */
+    Register m_xor_output{};
+
+    /**
+     * \brief Preprocess the calculation initial remainder.
+     *
+     * \param[in] initial_remainder The calculation initial remainder.
+     * \param[in] lookup_table The calculation lookup table.
+     *
+     * \return The preprocessed calculation initial remainder.
+     */
+    static constexpr auto preprocess_initial_remainder(
+        Register                            initial_remainder,
+        Byte_Lookup_Table<Register> const & lookup_table ) noexcept
+    {
+        auto remainder = initial_remainder;
```

Comment:
This variable should be renamed to reflect that it will contain the preprocessed initial remainder, not the specified initial remainder.